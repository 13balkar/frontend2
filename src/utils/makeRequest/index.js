import axios from 'axios';
import { BACKEND_URL } from '../../constants/apiEndPoints';

const makeRequest = async (apiEndPoint, dynamicConfig) => {
  try {
    const requestConfig = {
      url: `${BACKEND_URL}${apiEndPoint.url}`,
      method: apiEndPoint.method,
      ...dynamicConfig
    };
    console.log(requestConfig);
    const { data } = await axios(requestConfig);
    return data;
  } catch (error) {
    console.log('error', error);
    return error;
  }
};

export default makeRequest;
