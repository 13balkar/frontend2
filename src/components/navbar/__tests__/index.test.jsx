import * as React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../';
import makeRequest from '../../../utils/makeRequest';
jest.mock('../../../utils/makeRequest');
jest.mock('react-router-dom', () => ({
  useNavigate: () => jest.fn()
}));

describe('Navbar', () => {
  it('should render the navbar', () => {
    const setView = jest.fn();
    const { asFragment } = render(<Navbar view={true} setView={setView} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render the navbar with names', () => {
    const setView = jest.fn();
    makeRequest.mockResolvedValueOnce(['name1', 'name2']);
    const { asFragment } = render(<Navbar view={true} setView={setView} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
