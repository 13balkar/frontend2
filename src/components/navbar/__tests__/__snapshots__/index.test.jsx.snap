// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should render the homepage 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="heading"
    >
      <h1>
        CMS+
      </h1>
    </div>
    <div
      class="list"
    >
      <p>
        COLLECTION TYPES
      </p>
    </div>
    <div
      class="builder"
    >
      <p>
        CONTENT TYPE BUILDER
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar should render the navbar 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="heading"
    >
      <h1>
        CMS+
      </h1>
    </div>
    <div
      class="list"
    >
      <p>
        COLLECTION TYPES
      </p>
    </div>
    <div
      class="builder"
    >
      <p>
        CONTENT TYPE BUILDER
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Navbar should render the navbar with names 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="heading"
    >
      <h1>
        CMS+
      </h1>
    </div>
    <div
      class="list"
    >
      <p>
        COLLECTION TYPES
      </p>
    </div>
    <div
      class="builder"
    >
      <p>
        CONTENT TYPE BUILDER
      </p>
    </div>
  </div>
</DocumentFragment>
`;
